chainstate/src/detail/chainstateref.rs

```diff
@@ -127,8 +133,11 @@ impl<'a, S: BlockchainStorageRead, O: OrphanBlocks> ChainstateRef<'a, S, O> {
         (self.time_getter)()
     }
 
-    pub fn get_best_block_id(&self) -> Result<Option<Id<Block>>, PropertyQueryError> {
-        self.db_tx.get_best_block_id().map_err(PropertyQueryError::from)
+    pub fn get_best_block_id(&self) -> Result<Id<GenBlock>, PropertyQueryError> {
+        self.db_tx
+            .get_best_block_id()
+            .map_err(PropertyQueryError::from)
+            .map(|bid| bid.expect("Best block ID not initialized"))
```

Comment:
This is not strictly related to genesis/block separation. Just asserts that after chainstate has been initialised, the best block ID should always be available.